tests/apollo/test_skvbc_checkpoints.py

```diff
@@ -67,3 +66,65 @@ async def test_checkpoint_creation(self, bft_network):
         checkpoint_after = await bft_network.wait_for_checkpoint(replica_id=0)
 
         self.assertEqual(checkpoint_after, 1 + checkpoint_before)
+
+    @with_trio
+    @with_bft_network(start_replica_cmd)
+    async def test_checkpoint_creation_with_f_replicas_down(self, bft_network):
+        """
+        Here we bring down a total of f replicas, verify checkpoint creation in that scenario
+
+        1) Given a BFT network, we make sure all nodes are up
+        2) Crash f replicas, excluding the current primary
+        3) Send sufficient number of client requests to trigger checkpoint protocol
+        4) Make sure checkpoint is created and agreed upon
+        """
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+
+        n = bft_network.config.n
+        f = bft_network.config.f
+        c = bft_network.config.c
+
+        self.assertEqual(len(bft_network.procs), n,
+                         "Make sure all replicas are up initially.")
+        current_primary = await bft_network.get_current_primary()
+
+        checkpoint_before = await bft_network.wait_for_checkpoint(replica_id=current_primary)
+
+        crashed_replicas = await self._crash_replicas_without_primary(
+            bft_network=bft_network,
+            nb_crashing=f,
+            except_replicas={current_primary}
```

Comment:
You shouldn't have to pass the current primary here. Given the name of the function, you should just have it exclude the current primary.

Reply:
Makes sense. I will update it.

Reply:
Updated it Andrew, and added another test-case.

Reply:
The latter test seems to be a superset of the first. For runtime purposes, maybe just use that one?

Reply:
Yep, updated it.